File: from-0301-to-0400/0367-valid-perfect-square/solution.go
func isPerfectSquare(num int) bool {
    if num == 1 {
        return true
    }
    left, right := 1, int(num / 2)
    for left <= right {
        mid := left + (right - left) / 2
        m2 := mid * mid
        if m2 == num {
            return true
        } else if m2 > num {
            right = mid - 1
        } else {
            left = mid + 1
        }
    }
    return false
}

File: from-0301-to-0400/0367-valid-perfect-square/solution.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        if num == 1:
            return True

        left, right = 1, num // 2
        while left <= right:
            mid = (left + right) // 2
            m2 = mid * mid
            if m2 == num:
                return True
            elif m2 > num:
                right = mid - 1
            else:
                left = mid + 1

        return False 